Sources/Client/Model/Channel+Requests.swift

```diff
@@ -0,0 +1,457 @@
+                response.channel.add(messagesToDatabase: response.messages)
+            }
+        }
+        
+        return Client.shared.request(endpoint: .channel(channelQuery), fullCompletion)
+    }
+    
+    /// Loads the initial channel state and watches for changes.
+    /// - Parameters:
+    ///   - options: an additional channel options, e.g. `.presence`
+    ///   - completion: a completion block with `ChannelResponse`.
+    @discardableResult
+    func watch(options: QueryOptions = [], _ completion: @escaping Client.Completion<ChannelResponse>) -> URLSessionTask {
+        var options = options
+        
+        if !options.contains(.watch) {
+            options = options.union(.watch)
+        }
+        
+        return query(options: .watch, completion)
+    }
+    
+    /// Stop watching the channel for a state changes.
+    /// - Parameter completion: an empty completion block.
+    @discardableResult
+    func stopWatching(_ completion: @escaping Client.Completion<EmptyData> = { _ in }) -> URLSessionTask {
+        return Client.shared.request(endpoint: .stopWatching(self), completion)
+    }
+    
+    /// Hide the channel from queryChannels for the user until a message is added.
+    /// - Parameters:
+    ///   - user: the current user.
+    ///   - clearHistory: checks if needs to remove a message history of the channel.
+    ///   - completion: an empty completion block.
+    @discardableResult
+    func hide(for user: User? = User.current,
+              clearHistory: Bool = false,
+              _ completion: @escaping Client.Completion<EmptyData> = { _ in }) -> URLSessionTask {
+        let completion = doAfter(completion) { [weak self] _ in
+            self?.stopWatching()
+        }
+        
+        return Client.shared.request(endpoint: .hideChannel(self, user, clearHistory), completion)
+    }
+    
+    /// Removes the hidden status for a channel.
+    /// - Parameters:
+    ///   - user: the current user.
+    ///   - completion: an empty completion block.
+    @discardableResult
+    func show(for user: User? = User.current, _ completion: @escaping Client.Completion<EmptyData> = { _ in }) -> URLSessionTask {
+        guard let user = user else {
+            completion(.failure(.emptyUser))
+            return .empty
+        }
+        
+        return Client.shared.request(endpoint: .showChannel(self, user), completion)
+    }
+    
+    /// Update channel data.
+    /// - Parameters:
+    ///   - name: a channel name.
+    ///   - imageURL: an image URL.
+    ///   - extraData: a custom extra data.
+    ///   - completion: a completion block with `ChannelResponse`.
+    @discardableResult
+    func update(name: String? = nil,
+                imageURL: URL? = nil,
+                extraData: Codable? = nil,
+                _ completion: @escaping Client.Completion<ChannelResponse>) -> URLSessionTask {
+        var changed = false
+        
+        if let name = name, !name.isEmpty {
```

Comment:
Are we going to remove these check to let server handle these?

Reply:
It's not about server check. The method does an optional update. You can update only name or only imageURL and for keeping other values no need to add a parameter in the call.